docs/articles/documentation/test-api/selecting-page-elements/selectors.md

```diff
@@ -123,221 +131,311 @@ You can initialize a selector with any of these objects.
----------------------- | -------- | ------------------------------------------------------------------
-`init`                 | Function &#124; String &#124; Selector &#124; Snapshot &#124; Promise | Identifies a DOM node to be selected. See [Selector Initializers](#selector-initializers).
-`options`&#160;*(optional)* | Object   | See [Options](selector-options.md).
+Note that selector property getters are asynchronous.
 
-The following example shows how to get a DOM element by ID with `t.select`.
+### Obtain Element State
+
+Selectors and promises returned by selectors expose API to get state (size, position, classes, etc.) of matched element.
+See [DOM Node State](./dom-node-state.md). Note that these methods and property getters
+are asynchronous so you can obtain element's property from a browser in the following way:
 
 ```js
+const headerText = await Selector('#header').textConent;
+```
+
+For example:
+
+```js
+
+import { Selector } from 'testcafe';
+
 fixture `My fixture`
-    .page `http://www.example.com/`;
+    .page('http://devexpress.github.io/testcafe/example/');
 
-test('My Test', async t => {
-    const header = await t.select(() => document.getElementById('header'));
+const windowsInput = Selector('#windows');
+
+test('Obtain Element State', async t => {
+    await t.click(windowsInput);
+
+    const windowsInputChecked = await windowsInput.checked; // returns true
 });
-```
 
-## Using Selectors to Define Action Targets
+ ```
 
-You can pass selectors to [test actions](../actions/index.md) to use the returned element as the action target.
+#### DOM Node Snapshot
+
+If you need to get all state properties of the DOM element at once call the selector
+with the `await` keyword like you would do with regular async functions.
+It returns a *DOM Node Snapshot* that contains information about the node's size, position, classes, parent and child nodes, etc.
+It exposes [API](dom-node-state.md) that is similar to DOM objects.
 
 ```js
+
 import { Selector } from 'testcafe';
 
 fixture `My fixture`
-    .page `http://www.example.com/`;
+    .page `http://devexpress.github.io/testcafe/example/`;
 
-const lastItem = Selector(() => document.querySelector('.toc-item:last-child'));
+test('DOM Node Snapshot', async t => {
+    const sliderHandle = await Selector('#slider').child('span');
 
-test('My Test', async t => {
-    await t.click(lastItem);
+    await t
+        .expect(sliderHandle.hasClass('ui-slider-handle')).ok()
+        .expect(sliderHandle.childElementCount).eql(0)
+        .expect(sliderHandle.visible).ok();
 });
+
 ```
 
-In this instance, the selector will be called with no arguments.
+Note that if a selector initializer has several matching DOM nodes on the page,
+the selector returns the first node from the matching set.
 
-You can also pass a promise returned by a selector if you need to call it with arguments.
+But it's not quite good practice to pass DOM Node Snapshot's properties
```

Comment:
This should be highlighted as a note and use `It's not recommended to pass..` instead of  `But it's not quite good practice`